week6/Filmer/app/src/main/java/com/tahhan/filmer/view/DetailsFragment.kt

```diff
@@ -34,51 +30,44 @@ class DetailsFragment : Fragment() {
     private var isSavedLiveData = MutableLiveData<Boolean>()
 
     // Inflate the layout for this fragment
-
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-
         return inflater.inflate(R.layout.fragment_details, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        movieViewModel = ViewModelProviders.of(this).get(MovieViewModel::class.java)
+        movieViewModel = ViewModelProvider(this).get(MovieViewModel::class.java)
         arguments?.let {
             args = DetailsFragmentArgs.fromBundle(it)
 
         }
         populateData()
         movie = movieLiveData.value
         fab.setOnClickListener {
-            GlobalScope.launch(Dispatchers.IO) {
-                if (isSavedLiveData.value == false) {
-
-                    movieViewModel.insertMovie(movie!!)
-                    Snackbar.make(
-                        view.rootView,
-                        getString(R.string.movie_inserted),
-                        Snackbar.LENGTH_SHORT
-                    ).show()
-                    fab.setImageDrawable(resources.getDrawable(R.drawable.delete))
-                    isSavedLiveData.postValue(true)
-
-
-                } else {
-                    movieViewModel.removeMovie(movie!!)
-                    Snackbar.make(
-                        view.rootView,
-                        getString(R.string.movie_removed),
-                        Snackbar.LENGTH_SHORT
-                    ).show()
-                    fab.setImageDrawable(resources.getDrawable(R.drawable.heart))
-                    isSavedLiveData.postValue(false)
-                }
-
-
+            if (isSavedLiveData.value == false) {
+                movieViewModel.insertMovie(movie!!)
+                Toast.makeText(
+                    view.context,
+                    getString(R.string.movie_inserted),
+                    Toast.LENGTH_SHORT
+                ).show()
+                fab.setImageDrawable(resources.getDrawable(R.drawable.delete))
+                isSavedLiveData.postValue(true)
+
+            } else {
+                movieViewModel.removeMovie(movie!!)
```

Comment:
Split these up in different functions in the future. It doesn't make sense to have a 50-100+ line of codes in the onCreateView functions.